posthog/temporal/batch_exports/squash_person_overrides.py

```diff
@@ -15,76 +13,61 @@
 
 EPOCH = datetime(1970, 1, 1, 0, 0, tzinfo=timezone.utc)
 
-SELECT_PERSONS_TO_DELETE_QUERY = """
-SELECT
-    team_id,
-    old_person_id,
-    override_person_id,
-    max(created_at) AS latest_created_at,
-    max(version) AS latest_version,
-    min(oldest_event) AS oldest_event_at
-FROM
-    {database}.person_overrides
-WHERE
-    created_at <= %(latest_created_at)s
-GROUP BY
-    team_id, old_person_id, override_person_id
-"""
-
-SELECT_LATEST_CREATED_AT_QUERY = """
-SELECT
-    max(created_at)
-FROM {database}.person_overrides;
-"""
 
 CREATE_DICTIONARY_QUERY = """
 CREATE OR REPLACE DICTIONARY {database}.{dictionary_name} ON CLUSTER {cluster_name} (
-    `team_id` INT,
-    `old_person_id` UUID,
-    `override_person_id` UUID
+    `team_id` Int64,
+    `distinct_id` String,
+    `person_id` UUID
 )
-PRIMARY KEY team_id, old_person_id
-SOURCE(CLICKHOUSE(USER '{user}' PASSWORD '{password}' TABLE 'person_overrides' DB '{database}'))
+PRIMARY KEY team_id, distinct_id
+SOURCE(CLICKHOUSE(USER '{user}' PASSWORD '{password}' TABLE 'person_distinct_id_overrides' DB '{database}'))
 LAYOUT(complex_key_hashed())
-LIFETIME(0)
+LIFETIME(MIN 0 MAX 0)
+"""
+
+RELOAD_DICTIONARY_QUERY = """
+SYSTEM RELOAD DICTIONARY {database}.{dictionary_name}
 """
 
 SQUASH_EVENTS_QUERY = """
 ALTER TABLE
     {database}.sharded_events
 UPDATE
-    person_id = dictGet('{database}.{dictionary_name}', 'override_person_id', (toInt32(team_id), person_id))
+    person_id = dictGet('{database}.{dictionary_name}', 'person_id', (team_id, distinct_id))
 IN PARTITION
     %(partition_id)s
 WHERE
-    dictHas('{database}.{dictionary_name}', (toInt32(team_id), person_id))
-    {team_id_filter}
-    AND created_at <= %(latest_created_at)s;
+    dictHas('{database}.{dictionary_name}', (team_id, distinct_id))
+    {in_team_ids}
 """
 
 DROP_DICTIONARY_QUERY = """
 DROP DICTIONARY {database}.{dictionary_name};
 """
 
-DELETE_SQUASHED_PERSON_OVERRIDES_QUERY = """
-ALTER TABLE
-    {database}.person_overrides
-DELETE WHERE
-    old_person_id IN %(old_person_ids)s
-    AND created_at <= %(latest_created_at)s;
-"""
-
-SELECT_CREATED_AT_FOR_PERSON_EVENT_QUERY = """
+CREATE_JOIN_TABLE_FOR_DELETES_QUERY = """
+CREATE TABLE {database}.person_overrides_to_delete
```

Comment:
Yeah, haven't thought about multiple simultaneous squash workflows. It could get a bit nasty if too many mutations start running at the same time, so I don't think we want to run many anyways.

But I can review and see if we can easily support this.